code/run_templates/etl_sub_dataset_by_filter/create_subdataset.py

```diff
@@ -76,10 +77,33 @@ def create_subdataset(organization_id, workspace_id, parent_dataset_id, subdatas
     except cosmotech_api.ApiException as e:
         LOGGER.error("Failed to create subdataset: %s\n" % e)
         raise e
+    subdataset_id = subdataset["id"]
+    LOGGER.info(f'Sub dataset created, its id is "{subdataset_id}"')
+
+    time.sleep(2)  # Delay first status query to make sure back-end had time to init (see #SDCOSMO-1768 for example)
+    LOGGER.info("Starting status polling...")
+    ingestion_status = "PENDING"
+    max_polling_count = 20
+    polling_count = 0
+    while polling_count < max_polling_count:
+        polling_count += 1
+
+        ingestion_status = api["dataset"].get_dataset_twingraph_status(organization_id, subdataset_id)
```

Comment:
does this impose some maximum on the possible size of a subdataset, e.g. if it is too large or too slow? I guess we cannot do anything other than raise an error 🤷🏾, but if we have an order of magnitude in mind we could document this

Reply:
Good idea, yet I don't know what would be a reasonable maximum time for twincache ingestion, I just added this threshold to prevent runners from running indefinitely, because datasets may get stuck in PENDING state sometimes :smile: 
Also calling the number of polling calls like I did is not precise, because it does not take into account the time of the API response to the "get status" call. A future improvement could be to measure the difference between the current time and the polling start time to be more precise.

Reply:
@csm-thu LGTM